lib/src/async/stream_buffer.dart

```diff
@@ -173,12 +170,10 @@ class StreamBuffer<T> implements StreamConsumer<List<T>> {
 
   @override
   Future close() {
-    Future ret;
-    if (_sub != null) {
-      ret = _sub.cancel();
-      _sub = null;
-    }
-    return ret ?? Future.value();
+    Future? ret;
+    ret = _sub?.cancel();
```

Comment:
The previous 2 lines can be combined into one and `ret` can be made `final`.

Reply:
Thanks for spotting -- was done locally and forgot to re-push!